src/Algebra/Graph/Relation/Symmetric.hs

```diff
@@ -11,27 +11,497 @@
+
+-- | The number of vertices in a graph.
+-- Complexity: /O(1)/ time.
+--
+-- @
+-- vertexCount 'empty'             ==  0
+-- vertexCount ('vertex' x)        ==  1
+-- vertexCount                   ==  'length' . 'vertexList'
+-- vertexCount x \< vertexCount y ==> x \< y
+-- @
+vertexCount :: SymmetricRelation a -> Int
+vertexCount = R.vertexCount . fromSymmetric
+
+-- | The number of edges in a graph.
+-- Complexity: /O(1)/ time.
+--
+-- @
+-- edgeCount 'empty'      == 0
+-- edgeCount ('vertex' x) == 0
+-- edgeCount ('edge' x y) == 2
+-- edgeCount            == 'length' . 'edgeList'
+-- @
+edgeCount :: SymmetricRelation a -> Int
+edgeCount = R.edgeCount . fromSymmetric
+
+-- | The sorted list of vertices of a given graph.
+-- Complexity: /O(n)/ time and memory.
+--
+-- @
+-- vertexList 'empty'      == []
+-- vertexList ('vertex' x) == [x]
+-- vertexList . 'vertices' == 'Data.List.nub' . 'Data.List.sort'
+-- @
+vertexList :: SymmetricRelation a -> [a]
+vertexList = R.vertexList . fromSymmetric
+
+-- | The sorted list of edges of a graph.
+-- Complexity: /O(n + m)/ time and /O(m)/ memory.
+--
+-- @
+-- edgeList 'empty'          == []
+-- edgeList ('vertex' x)     == []
+-- edgeList ('edge' x y)     == [(x,y), (y,x)]
+-- edgeList ('star' 2 [3,1]) == [(2,1), (1,2), (2,3), (3,2)]
+-- @
+edgeList :: SymmetricRelation a -> [(a, a)]
+edgeList = R.edgeList . fromSymmetric
+
+-- | The set of vertices of a given graph.
+-- Complexity: /O(1)/ time.
+--
+-- @
+-- vertexSet 'empty'      == Set.'Set.empty'
+-- vertexSet . 'vertex'   == Set.'Set.singleton'
+-- vertexSet . 'vertices' == Set.'Set.fromList'
+-- @
+vertexSet :: SymmetricRelation a -> Set.Set a
+vertexSet = R.vertexSet . fromSymmetric
+
+-- | The set of edges of a given graph.
+-- Complexity: /O(1)/ time.
+--
+-- @
+-- edgeSet 'empty'      == Set.'Set.empty'
+-- edgeSet ('vertex' x) == Set.'Set.empty'
+-- edgeSet ('edge' x y) == Set.'Set.fromList' [(x,y), (y,x)]
+-- @
+edgeSet :: SymmetricRelation a -> Set.Set (a, a)
+edgeSet = R.edgeSet . fromSymmetric
+
+-- | The sorted /adjacency list/ of a graph.
+-- Complexity: /O(n + m)/ time and /O(m)/ memory.
+--
+-- @
+-- adjacencyList 'empty'          == []
+-- adjacencyList ('vertex' x)     == [(x, [])]
+-- adjacencyList ('edge' 1 2)     == [(1, [2]), (2, [1])]
+-- adjacencyList ('star' 2 [3,1]) == [(1, [2]), (2, [1,3]), (3, [2])]
+-- 'stars' . adjacencyList        == id
+-- @
+adjacencyList :: Eq a => SymmetricRelation a -> [(a, [a])]
+adjacencyList = R.adjacencyList . fromSymmetric
+
+-- | The /preset/ of an element @x@ is the set of elements that are related to
+-- it on the /left/, i.e. @preSet x == { a | aRx }@. In the context of directed
+-- graphs, this corresponds to the set of /direct predecessors/ of vertex @x@.
+-- Complexity: /O(n + m)/ time and /O(n)/ memory.
+--
+-- @
+-- preSet x 'empty'      == Set.'Set.empty'
+-- preSet x ('vertex' x) == Set.'Set.empty'
+-- preSet 1 ('edge' 1 2) == Set.'Set.singleton' 2
```

Comment:
I think we should delete `preSet` and `postSet` methods, because they are not relevant to undirected graphs.

Reply:
Alright!